src/lexer.coffee

```diff
@@ -761,6 +776,11 @@ exports.Lexer = class Lexer
     [..., token] = @tokens
     token?[1]
 
+  # Get the previous token in the token stream.
+  prev: ->
+    [..., token] = @tokens
+    token if token?
```

Comment:
I think those two lines can be replaced with `@tokens[@tokens.length - 1]` - if there's no token, it'll be `undefined`